### PR TITLE
Bump upstreams to 2022.09

### PR DESCRIPTION
**Pull request by Tobianas**
- odlparent-11.0.0
- infrautils-4.0.0
- yangtools-9.0.0
- mdsal-10.0.0
- controller-6.0.0
- aaa-0.16.0
- netconf-4.0.0
- bgpcep-0.18.0

ID: 1831

### CODE REVIEW

**Review by PeterSuna — CHANGES_REQUESTED** [review]

File: lighty-modules/lighty-netconf-sb/pom.xml [hunk 1]
```diff
@@ -83,6 +83,18 @@
...
+        <artifactId>rfc8342-ietf-origin</artifactId>
+          <version>10.0.0</version>
+        <scope>test</scope>
+      </dependency>
+        <dependency>
+            <groupId>org.opendaylight.netconf</groupId>
+            <artifactId>ietf-netconf-with-defaults</artifactId>
+            <version>4.0.0-SNAPSHOT</version>
+            <scope>test</scope>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (Tobianas) [reply lighty-modules/lighty-netconf-sb/pom.xml:93]: the reason is that their placement got refractored
https://github.com/opendaylight/netconf/commit/f5199ed366896729160b9f1cafc597be81b5226e

File: lighty-core/dependency-versions/pom.xml [hunk 2]
```diff
@@ -264,9 +264,35 @@
                 <version>20220320</version>
                 <scope>test</scope>
             </dependency>
+            <dependency>
```
[the remark this answers is not in the document]
Comment (Tobianas) [reply lighty-core/dependency-versions/pom.xml:267]: replaced:
https://github.com/PANTHEONtech/lighty/pull/1024/commits/3cf138229951dd82f001a22892821f6281bc17a5

File: lighty-modules/lighty-aaa-aggregator/lighty-aaa/pom.xml [hunk 3]
```diff
@@ -68,6 +68,11 @@
             <artifactId>mockito-inline</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.opendaylight.aaa.web</groupId>
+            <artifactId>servlet-jersey2</artifactId>
+            <version>0.16.0-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (Tobianas) [reply lighty-modules/lighty-aaa-aggregator/lighty-aaa/pom.xml]: Removed in:
https://github.com/PANTHEONtech/lighty/pull/1024/commits/de2737c5e87c55607c8035c0d5f2443595365029

File: lighty-modules/lighty-netconf-sb/pom.xml [hunk 4]
```diff
@@ -83,18 +79,38 @@
...
+        <dependency>
+            <groupId>org.opendaylight.netconf.model</groupId>
+            <artifactId>rfc6022</artifactId>
+            <version>4.0.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.opendaylight.netconf.model</groupId>
+            <artifactId>rfc8526</artifactId>
+            <version>4.0.0</version>
+        </dependency>
```
Comment (PeterSuna) [inline lighty-modules/lighty-netconf-sb/pom.xml:113]: Remove rfc8526 dependency. It is duplicated with line 58.

File: lighty-core/dependency-versions/pom.xml [hunk 5]
```diff
@@ -264,9 +264,15 @@
...
+                <groupId>org.opendaylight.netconf</groupId>
+                <artifactId>netconf-parent</artifactId>
+                <version>4.0.0</version>
+            </dependency>
         </dependencies>
     </dependencyManagement>
 
+
```
Comment (PeterSuna) [inline lighty-core/dependency-versions/pom.xml:275]: Remove this new line

**Review by PeterSuna — CHANGES_REQUESTED** [review]
Please drop commits with removing SemVer and excluding gNMI and rebase this to #1045 PR which partially solves this problem.

File: lighty-modules/lighty-netconf-sb/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tobianas) [reply lighty-modules/lighty-netconf-sb/pom.xml:85]: versions removed

File: lighty-modules/lighty-aaa-aggregator/lighty-aaa/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Tobianas) [reply lighty-modules/lighty-aaa-aggregator/lighty-aaa/pom.xml]: version removed and moved

File: lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/impl/LightyControllerImpl.java [hunk 6]
```diff
@@ -89,7 +89,7 @@
 import org.opendaylight.mdsal.binding.dom.codec.api.BindingCodecTreeFactory;
 import org.opendaylight.mdsal.binding.dom.codec.api.BindingNormalizedNodeSerializer;
 import org.opendaylight.mdsal.binding.dom.codec.impl.BindingCodecContext;
-import org.opendaylight.mdsal.binding.dom.codec.impl.DefaultBindingCodecTreeFactory;
+import org.opendaylight.mdsal.binding.dom.codec.impl.di.DefaultBindingCodecTreeFactory;
```
[the remark this answers is not in the document]
Comment (Tobianas) [reply lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/impl/LightyControllerImpl.java:92]: Added to separate commit

File: lighty-core/dependency-versions/pom.xml [hunk 7]
```diff
@@ -53,49 +53,49 @@
...
-                <version>9.0.4</version>
+                <version>10.0.0</version>
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
             <dependency>
                 <groupId>org.opendaylight.netconf</groupId>
                 <artifactId>netconf-artifacts</artifactId>
-                <version>3.0.5</version>
+                <version>4.0.0-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (Tobianas) [reply lighty-core/dependency-versions/pom.xml]: Changed to 4.0.0

File: lighty-modules/lighty-netconf-sb/pom.xml (hunk 4 above)
Comment (Tobianas) [reply lighty-modules/lighty-netconf-sb/pom.xml]: Removed

File: lighty-modules/lighty-netconf-sb/pom.xml [hunk 8]
```diff
@@ -46,20 +46,16 @@
             <artifactId>callhome-provider</artifactId>
         </dependency>
         <dependency>
-            <groupId>org.opendaylight.netconf</groupId>
-            <artifactId>ietf-netconf</artifactId>
+            <groupId>org.opendaylight.netconf.model</groupId>
```
[the remark this answers is not in the document]
Comment (Tobianas) [reply lighty-modules/lighty-netconf-sb/pom.xml:49]: Unudený dependencies removed

File: lighty-core/dependency-versions/pom.xml (hunk 5 above)
Comment (Tobianas) [reply lighty-core/dependency-versions/pom.xml]: Should be removed

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfConfigUtils.java [hunk 9]
```diff
@@ -57,8 +57,6 @@ public final class NetconfConfigUtils {
                     .$YangModuleInfoImpl.getInstance(),
             org.opendaylight.yang.gen.v1.urn.ietf.params.xml.ns.yang.ietf.netconf.monitoring.rev101004
                     .$YangModuleInfoImpl.getInstance(),
-            org.opendaylight.yang.gen.v1.urn.ietf.params.xml.ns.yang.ietf.netconf.monitoring.extension.rev131210
```
[the remark this answers is not in the document]
Comment (Tobianas) [reply lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfConfigUtils.java:60]: Model replaced and changed tests based on new model

File: lighty-modules/lighty-netconf-sb/pom.xml [hunk 10]
```diff
@@ -83,6 +71,19 @@
             <artifactId>singlenode-configuration</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (PeterSuna) [inline lighty-modules/lighty-netconf-sb/pom.xml:74]: Fix dependencies for correct ones that should have been replaced.

   ietf-netconf -> rfc6241
 https://git.opendaylight.org/gerrit/c/netconf/+/101841/11/model/rfc6241/pom.xml
 
 ietf-netconf-monitoring-> rfc6022
 https://git.opendaylight.org/gerrit/c/netconf/+/101841/11/model/rfc6022/pom.xml#20
 
 ietf-netconf-nmda -> rfc8526
 https://git.opendaylight.org/gerrit/c/netconf/+/101841/11/model/rfc8526/pom.xml#20
 
 netconf-monitoring-extension -> odl-netconf-monitoring 
 https://git.opendaylight.org/gerrit/c/netconf/+/101857

In the commit message, write which dependency was replaced by which one.

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-docker/example-config/configuration.json [hunk 11]
```diff
@@ -39,7 +39,7 @@
                 { "usedBy": "NETCONF", "name": "netconf-keystore", "revision": "2017-10-17", "nameSpace": "urn:opendaylight:netconf:keystore"},
                 { "usedBy": "NETCONF", "name": "netconf-node-topology", "revision": "2015-01-14", "nameSpace": "urn:opendaylight:netconf-node-topology"},
                 { "usedBy": "NETCONF", "name":"ietf-netconf", "revision":"2011-06-01", "nameSpace":"urn:ietf:params:xml:ns:netconf:base:1.0"},
-                { "usedBy": "NETCONF", "name": "ietf-netconf-monitoring-extension", "revision": "2013-12-10", "nameSpace": "urn:ietf:params:xml:ns:yang:ietf-netconf-monitoring-extension"},
+                { "usedBy": "NETCONF", "name": "ietf-netconf-nmda", "revision": "2019-01-07", "nameSpace": "urn:ietf:params:xml:ns:yang:ietf-netconf-nmda"},
```
Comment (PeterSuna) [inline lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-docker/example-config/configuration.json:42]: ietf-netconf-monitoring-extension was replaced by odl-netconf-monitoring.
 https://git.opendaylight.org/gerrit/c/netconf/+/101857
 
 But maybe it will not be required. Try to find out if we need this one:
 https://git.opendaylight.org/gerrit/c/netconf/+/101856/1

**Review by PeterSuna — CHANGES_REQUESTED** [review]

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/gnmi/GnmiCapabilitiesService.java [hunk 12]
```diff
@@ -41,20 +49,36 @@ public Gnmi.CapabilityResponse getResponse(final Gnmi.CapabilityRequest request)
...
-            .build();
+                .addAllSupportedModels(modelDataList)
+                .addAllSupportedEncodings(supportedEncodings)
+                .setGNMIVersion(GNMI_VERSION)
+                .build();
+    }
+
+    private Optional<SemVer> getSemVer(final Module module) {
+        if (module instanceof AbstractDeclaredEffectiveStatement) {
+            final var decStmt = ((AbstractDeclaredEffectiveStatement) module).getDeclared().declaredSubstatements();
```
Comment (ihrasko) [inline lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/gnmi/GnmiCapabilitiesService.java:72]: `module.asEffective`

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/gnmi/GnmiCapabilitiesService.java [hunk 13]
```diff
@@ -41,20 +49,36 @@ public Gnmi.CapabilityResponse getResponse(final Gnmi.CapabilityRequest request)
...
+            final var semVerStmt = ((List<? extends DeclaredStatement<?>>) decStmt).stream()
+                    .filter(UnrecognizedStatementImpl.class::isInstance)
+                    .map(UnrecognizedStatementImpl.class::cast)
+                    .filter(t -> t.statementDefinition().getStatementName().getLocalName().equals(OPENCONFIG_VERSION))
+                    .findAny();
+            if (semVerStmt.isPresent()) {
```
Comment (ihrasko) [inline lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/gnmi/GnmiCapabilitiesService.java:78]: `semVerStmt.map(..)`

File: lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/schema/loader/util/YangLoadModelUtil.java [hunk 14]
```diff
@@ -62,5 +72,18 @@ public String getModelBody() {
...
         return modelName;
     }
+
+    public static Optional<SemVer> getSemVer(final IRStatement stmt) {
+        SemVer foundSemver = null;
+        for (final var substatement : stmt.statements()) {
+            if (OPENCONFIG_VERSION.equals(substatement.keyword().identifier())) {
+                final var argument = substatement.argument();
+                if (argument instanceof Single) {
+                    foundSemver = SemVer.valueOf(((Single) argument).string());
```
Comment (ihrasko) [inline lighty-modules/lighty-gnmi/lighty-gnmi-sb/src/main/java/io/lighty/gnmi/southbound/schema/loader/util/YangLoadModelUtil.java:82]: `return`

File: lighty-core/dependency-versions/pom.xml [hunk 15]
```diff
@@ -81,7 +81,7 @@
             <dependency>
                 <groupId>org.opendaylight.netconf</groupId>
                 <artifactId>netconf-artifacts</artifactId>
-                <version>4.0.0-SNAPSHOT</version>
```
Comment (ihrasko) [inline lighty-core/dependency-versions/pom.xml]: Do it in separate commit please.

**Review by ihrasko — CHANGES_REQUESTED** [review]

File: lighty-core/dependency-versions/pom.xml (hunk 15 above)
Comment (Tobianas) [reply lighty-core/dependency-versions/pom.xml]: separated

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/gnmi/GnmiCapabilitiesService.java [hunk 16]
```diff
@@ -62,4 +68,17 @@ public Gnmi.CapabilityResponse getResponse(final Gnmi.CapabilityRequest request)
             .setGNMIVersion(GNMI_VERSION)
             .build();
     }
+
+    private Optional<SemVer> getSemVer(final Module module) {
```
Comment (PeterSuna) [inline lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/gnmi/GnmiCapabilitiesService.java:72]:
```java
    private Optional<SemVer> getSemVer(final Module module) {
            final var decStmt = module.asEffectiveStatement().getDeclared().declaredSubstatements();
            return ((List<? extends DeclaredStatement<?>>) decStmt).stream()
                    .filter(UnrecognizedStatementImpl.class::isInstance)
                    .map(UnrecognizedStatementImpl.class::cast)
                    .filter(t -> t.statementDefinition().getStatementName().equals(OC_VERSION))
                    .findAny()
                    .map(stmt -> SemVer.valueOf(stmt.argument().toString()));
    }
```

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfConfigUtils.java [hunk 17]
```diff
@@ -57,7 +57,7 @@ public final class NetconfConfigUtils {
...
+            org.opendaylight.yang.gen.v1.urn.ietf.params.xml.ns.yang.ietf.netconf.nmda.rev190107
```
Comment (PeterSuna) [inline lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfConfigUtils.java:60]: Remove NMDA model from default netconf models if this is typo or describe why is needed here.

**Review by PeterSuna — CHANGES_REQUESTED** [review]

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/pom.xml [hunk 18]
```diff
@@ -40,7 +40,11 @@
         <dependency>
             <groupId>io.lighty.modules</groupId>
             <artifactId>lighty-swagger</artifactId>
-        </dependency>        
+        </dependency>
+        <dependency>
+            <groupId>org.opendaylight.mdsal.binding.model.ietf</groupId>
+            <artifactId>rfc8342-ietf-origin</artifactId>
```
Comment (PeterSuna) [inline lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/pom.xml:46]: Please remove this dependency if it is not needed here. I was able to build an application without it. If it is required, move it to module which use it. This is example and should be here only dependencies required to run lighty example.

File: lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/gnmi/GnmiCapabilitiesService.java (hunk 16 above)
Comment (Tobianas) [reply lighty-modules/lighty-gnmi/lighty-gnmi-device-simulator/src/main/java/io/lighty/modules/gnmi/simulatordevice/gnmi/GnmiCapabilitiesService.java:72]: changed

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfConfigUtils.java (hunk 17 above)
Comment (Tobianas) [reply lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfConfigUtils.java:60]: Removed this model but not replaced with odl-netconf-monitoring because of issues with odl-netconf-monitoring dependency

File: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/pom.xml (hunk 18 above)
Comment (Tobianas) [reply lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-module/pom.xml:46]: removed

**Comment by ihrasko** [comment]
Completed in https://github.com/PANTHEONtech/lighty/pull/1056.
